File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.js
function solution(wallpaper) {
    const rowLen = wallpaper.length, colLen = wallpaper[0].length;
    const matrix = Array.from(wallpaper, (row) => Array.from(row, (char) => char === '.' ? 0 : 1))
    // [ 
    //   [ 0, 1, 0, 0, 0 ], 
    //   [ 0, 0, 1, 0, 0 ], 
    //   [ 0, 0, 0, 1, 0 ] 
    // ]

    // 가장 max 왼쪽 값, max 위쪽 값, max 오른쪽 값, max 아래 값을 찾아야 한다. => S(위, 왼), E(아래, 오)
    let left = colLen - 1, up = rowLen - 1, right = 0, down = 0;
    for (let row = 0; row < rowLen; row++) {
        for (col = 0; col < colLen; col++) {
            const value = matrix[row][col]
            if (value) { // 1
                up = Math.min(up, row)
                left = Math.min(left, col)
                down = Math.max(down, row + 1)
                right = Math.max(right, col + 1)
            }
        }
    }
    return [up, left, down, right]
}

File: 프로그래머스/1/161990. 바탕화면 정리/바탕화면 정리.py
def solution(wallpaper):
    answer = []
    
    top = len(wallpaper) - 1
    bottom = 0
    left = len(wallpaper[0]) - 1
    right = 0
    
    for rowIdx in range(len(wallpaper)):
        for colIdx in range(len(wallpaper[0])):
            if wallpaper[rowIdx][colIdx] == '#':
                if rowIdx < top:
                    top = rowIdx
                if rowIdx > bottom:
                    bottom = rowIdx
                if colIdx < left:
                    left = colIdx
                if colIdx > right:
                    right = colIdx
                
    return [top, left, bottom + 1, right + 1]